src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -944,8 +944,9 @@ private ImmutableHashSet<ISymbol> GetOrComputeGeneratedCodeSymbolsInTree(SyntaxT
         private ImmutableHashSet<ISymbol> ComputeGeneratedCodeSymbolsInTree(SyntaxTree tree, Compilation compilation, CancellationToken cancellationToken)
         {
             // PERF: Bail out early if file doesn't have "GeneratedCode" text.
-            var text = tree.GetText(cancellationToken).ToString();
-            if (!text.Contains("GeneratedCode"))
+            var walker = new GeneratedCodeTokenWalker();
```

Comment:
>GeneratedCodeTokenWalker [](start = 29, length = 24)

Do we need a specialized visitor or could we use existing helpers that look for nodes, like ```DescendantNodes``` etc.?

Reply:
`DescendantTokens` should work. I'll check this out.

Reply:
@AlekseyTs `DescendantTokens` produces the same final result, but has a non-trivial impact on CPU time (about 45 thread-seconds). Allocations were the same with both approaches. Let me know how you would like me to proceed.

Reply:
> DescendantTokens produces the same final result, but has a non-trivial impact on CPU time (about 45 thread-seconds). Allocations were the same with both approaches. Let me know how you would like me to proceed

I'll let you make the choice.

---
In reply to: [371547413](https://github.com/dotnet/roslyn/pull/41209#discussion_r371547413) [](ancestors = 371547413)

Reply:
I'll go with the walker for now. I expect `DescendantTokens` will end up getting some performance work, but not before it shows up as a top contributor to overall analyzer performance.